mig/shared/install.py

```diff
@@ -485,34 +498,53 @@ def generate_confs(
     ca_fqdn='',
     ca_user='mig-ca',
     ca_smtp='localhost',
+    _getcwd=os.getcwd,
     _getpwnam=pwd.getpwnam,
+    _prepare=None,
+    _writefiles=None,
+    _instructions=None,
 ):
     """Generate Apache and MiG server confs with specified variables"""
 
+    # TODO: override in signature as a non-functional follow-up change
+    if _prepare is None:
+        _prepare = _generate_confs_prepare
+    if _writefiles is None:
+        _writefiles = _generate_confs_writefiles
+    if _instructions is None:
+        _instructions = _generate_confs_instructions
+
     # Read out dictionary of args with defaults and overrides
 
     thelocals = locals()
     expanded = {k: v for k, v in thelocals.items() if k not in
                 _GENERATE_CONFS_NOFORWARD_KEYS}
 
     # expand any directory path specific as "auto" relative to CWD
+    thecwd = _getcwd()
+
     if source == keyword_auto:
-        template_path = os.path.dirname(sys.argv[0])
+        # use the templates from this copy of the code tree
+        template_dir = os.path.join(MIG_BASE, "mig/install")
     else:
-        template_path = source
+        # construct a path using the supplied value made absolute
+        template_dir = abspath(source, start=thecwd)
 
     if destination == keyword_auto:
-        destination_link = os.path.dirname(sys.argv[0])
+        # write output into a confs folder within the CWD
+        destination = os.path.join(thecwd, 'confs')
     else:
-        destination_link = destination
+        # construct a path from the supplied value made absolute
+        destination = abspath(destination, start=thecwd)
 
     # expand any user information marked as "auto" based on the environment
     if user == keyword_auto:
         user = pwd.getpwuid(os.getuid())[0]
     if group == keyword_auto:
         group = grp.getgrgid(os.getgid())[0]
 
-    # finalize a destination path up-front
+    # finalize destination paths up-front
+    destination_link = destination
```

Comment:
I'll probably move these destination lines up above the user+group lines during merge to keep similar bits in one place

Reply:
yeah, agreed - was just trying to minimize the diff as much as possible.